botnim/kb/download_sources.py

```diff
@@ -0,0 +1,98 @@
+from pathlib import Path
+import os
+import requests
+import csv
+from io import StringIO
+import yaml
+from ..config import get_logger
+from typing import List, Tuple, BinaryIO
+import io
+
+logger = get_logger(__name__)
+
+def download_and_convert_spreadsheet(source_url: str, target_dir: Path, context_name: str) -> List[Tuple[str, BinaryIO, str]]:
+    """Download and convert Google Spreadsheet data to memory buffers
+    
+    Args:
+        source_url: URL of the Google Spreadsheet
+        target_dir: Directory to store files
+        context_name: Name of the context for file naming
+        
+    Returns:
+        List of tuples (filename, file_buffer, content_type) for entries
+    """
+    try:
+        logger.info(f"Downloading spreadsheet from {source_url}")
+        sheet_id = source_url.split('/d/')[1].split('/')[0]
+        url = f'https://docs.google.com/spreadsheets/d/{sheet_id}/export?format=csv'
+        
+        response = requests.get(url)
+        response.encoding = 'utf-8'
+        response.raise_for_status()
+        
+        logger.debug("Raw response text (first 500 chars):")
+        logger.debug(response.text[:500])
+        
+        # Create target directory if it doesn't exist
+        target_dir.mkdir(exist_ok=True, parents=True)
+        
+        csv_file = StringIO(response.text)
+        csv_reader = csv.reader(csv_file)
+        
+        headers = next(csv_reader)
+        headers = [h.strip() for h in headers if h.strip()]
+        
+        documents = []
+        
+        for i, columns in enumerate(csv_reader):
+            columns = [col.strip() for col in columns[:len(headers)]]
+            
+            if not columns[0]:  # Skip empty entries
+                continue
+                
+            entry = []
+            for header, value in zip(headers, columns):
+                if value:
+                    entry.append(f"{header}:\n{value}\n\n")
+                    
+            content = ''.join(entry)
+            if content.strip():
+                filename = f"{context_name}_{i+1:03d}.md"
+                file_path = target_dir / filename
```

Comment:
again, let's skip writing the actual files...

Reply:
removed